build_and_sync_requirements.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -xeo pipefail
+
+# This creates a hash from all the files that are used to generate the compiled
+# requirements file and then uses that hash to name the output file. This ensures
+# that the compiled requirements file always matches sources used to generate it.
+
+function cleanup() {
+    trap - EXIT SIGINT SIGTERM
+    if [ -f "${scratch_file}" ] ; then
+        rm -f "${scratch_file}"
+    fi
+}
+
+trap cleanup EXIT SIGINT SIGTERM
+
+scratch_file=$(mktemp)
+
+envsubst < requirements-dev.in > ${scratch_file}
+
+# also hash setup.py as it has the install_requires and extras
+SHA_SUM=$(sha1sum ${scratch_file} setup.py | head -c 40)
```

Comment:
sha1sum produces two lines with two items each, which are piped to head which takes the first 40 bytes only. Looking at CI runs, this appears to be fragile and it completely ignores the hash of setup.py.

May I propose this alternative?

`cat ${scratch_file} setup.py | sha1sum | awk '{print $1}'`

1. Concatenate both inputs
2. Pipe into sha1sum
3. Extract first word (the 40 character hash)

Reply:
🤦🏼‍♂️ thanks, that is a very odd default behavior of sha1sum.  I'll re-work